prov/shm/src/smr_msg.c

```diff
@@ -350,19 +294,18 @@ static ssize_t smr_generic_inject(struct fid_ep *ep_fid, const void *buf,
 		goto unlock;
 	}
 
-	cmd = ofi_cirque_next(smr_cmd_queue(peer_smr));
-	smr_generic_format(cmd, peer_id, op, tag, data, op_flags);
+	if (len <= SMR_MSG_DATA_LEN)
+		ret = smr_proto_ops[smr_src_inline](ep, peer_smr, id, peer_id,
+				op, tag, data, op_flags, FI_HMEM_SYSTEM, 0,
+				&msg_iov, 1, len, NULL);
+	else
+		ret = smr_proto_ops[smr_src_inject](ep, peer_smr, id, peer_id,
+				op, tag, data, op_flags, FI_HMEM_SYSTEM, 0,
+				&msg_iov, 1, len, NULL);
 
```

Comment:
Since all the arguments are the same, it would have less code duplication if proto is set by the if ... else ... statement and the function is called after that.